daemon/logger/ring.go

```diff
@@ -22,12 +22,12 @@ type RingLogger struct {
 }
 
 var (
-	_ SizedLogger = (*RingLogger)(nil)
+	_ SizedLogger = (*ringLogger)(nil)
 	_ LogReader   = (*ringWithReader)(nil)
```

Comment:
Actually wondering if (but for a  follow-up) `NewRingLogger` should explicitly return a `SizedLogger`, because that's what's implemented by both; https://github.com/moby/moby/blob/264801efb1f8676a6a4b1b6e023fbdbb56cf28e6/daemon/logger/ring.go#L53-L55

And perhaps assert that here as well for `ringWithReader`;

```go
	_ SizedLogger = (*ringWithReader)(nil)
```